### ISSUE TITLE
Check for invocations of asyncio.create_task without assignment-of-the-result

### ISSUE THREAD

**Issue by jayaddison** [PyCQA/pylint#8263]
### Current problem

A heisenbug condition described here: https://textual.textualize.io/blog/2023/02/11/the-heisenbug-lurking-in-your-async-code/

### Desired solution

Detect method calls to `asyncio.create_task` that do not assign the result to a variable.

**Comment by jayaddison** [comment]
(note: I'll likely make an attempt at implementing this within the next week or so.  it could be a slow development process)

**Comment by Pierre-Sassoulas** [comment]
There's a similar message here: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/useless-with-lock.html

Seems like something generic could be made, like ``thread-lost-at-creation`` or ``unassigned-thread-created`` ?

**Comment by jayaddison** [comment]
Excellent, thanks @Pierre-Sassoulas.  My initial sense would be not to adjust existing check names, but I'll think about it more.

By the way: searching around on GitHub, I see a bunch of codebases where `useless-with-lock` is ignored via comments for (I guess?) false-positives during [`threading.RLock`](https://docs.python.org/3/library/threading.html#threading.RLock) usage.

Perhaps another bug to file..

**Comment by Pierre-Sassoulas** [comment]
> I see a bunch of codebases where useless-with-lock is ignored via comments for (I guess?) false-positives during [threading.RLock](https://docs.python.org/3/library/threading.html#threading.RLock) usage.

We never had any report about it, so nice catch ! It could also be that some maintainers like to only enable explicitly (?) This would need some investigation.

**Comment by jayaddison** [comment]
A naive implementation of this check can be achieved by extending the [code relating to detecting `W0133`](https://github.com/PyCQA/pylint/blob/884034d342eeb3fd753733176ff99774f822f336/pylint/checkers/base/basic_checker.py#L452-L468):

```py
+        if name == "create_task" and expr.as_string().startswith("asyncio.create_task"):
+            self.add_message(
+                "unreferenced-task-creation", node=node, confidence=INFERENCE
+            )
```

That wouldn't catch all scenarios, though: currently I'm wondering about situations where a method might `return asyncio.create_task(...)` - where I guess type inference from the call site (plus a check to see whether assignment takes place) would be required.

(no reply required - some guidance would likely help me, but I should be able to figure things out myself given enough time)
